gulpfile.js

```diff
@@ -1,169 +1,122 @@
-gulp.task('copy', () => {
-	
-	return gulp.src(staticSrc)
-	.pipe(gulp.dest('dist/'))
-})
-
-/*  
- * SASS
- */
-gulp.task("sass", () => {
-
-	let out = gulp.src('src/scss/base.scss')
-		.pipe( $.cssGlobbing({
-			extensions: ['.scss']
-		})); 
-
-	// Create Sourmaps for develop
-	if (debug) {
-
-		return out.pipe($.sourcemaps.init())
-			.pipe($.sass({ style: 'compressed', sourcemap: true}))
-			.on('error', $.sass.logError)
-			.on('error', (err) => {
-				$.notify().write(err);
-			})
-			.pipe( $.autoprefixer({
-				browsers: ['last 2 versions','ie >= 9'],
-				cascade: false
-			}))
-			.pipe($.rename('site.min.css'))
-			.pipe($.sourcemaps.write('./'))
-			.pipe(gulp.dest('./dist/css'))
-			.pipe(browserSync.stream({match: '**/*.css'}));
-
-		}
-
-	// Remove sourcemaps and minify for production
-	else {
-		return out.pipe($.sass({ style: 'compressed'}))
-			.on('error', $.sass.logError)
-			.on('error', (err) => {
-				$.notify().write(err);
-			})
-			.pipe( $.autoprefixer({
-				browsers: ['last 2 versions','ie >= 9'],
-				cascade: false
-			}))
-			.pipe($.rename('site.min.css'))
-			.pipe(gulp.dest('./dist/css'));
-	}
+gulp.task('copy', () => gulp.src(STATIC_SRC)
+	.pipe(gulp.dest('dist/')));
 
+/**
+ * Sass to CSS build task
+ */
+gulp.task('sass', () => {
+	// get mode for conditional piping
+	const isDev = process.env.NODE_ENV === 'development';
+
+	return gulp.src('src/scss/base.scss')
+		.pipe(sassGlob())
+		.pipe(sass({ style: 'compressed' }))
+		.on('error', handleError)
+		.pipe(cssPrefixer(config.CSS_AUTOPREFIXER))
+		.pipe(rn('site.min.css'))
+		.pipe(gulp.dest('./dist/css'))
+		// conditionally stream compiled css to browserSync
+		.pipe(pipeIf(isDev, browserSync
+				.stream({ match: '**/*.css' }))
+		);
 });
 
-/*  
- * Javascript
+/**
+ * Bundle site js and modules
  */
 gulp.task('js', () => {
-	
-	// Development 
-	if (debug) {
-		return gulp.src('src/js/site.js')
-			.pipe($.sourcemaps.init())
-			.pipe($.browserify({
-				insertGlobals : true,
-				debug : debug
-			}))
-			.on('error', (err) => {
-				$.notify().write(err);
-			})
-			.pipe($.babel({
-				presets: ['es2015']
-			}))
-			.pipe($.sourcemaps.write('./'))
-			.pipe(gulp.dest('dist/js'))
-	}
-	// Production 
-	else {
-		return gulp.src('src/js/site.js')
-			.pipe($.browserify({
-				insertGlobals : true,
-				debug : debug
-			}))
-			.on('error', (err) => {
-				$.notify().write(err);
-			})
-			.pipe($.babel({
-				presets: ['es2015']
-			}))
-			.pipe(gulp.dest('dist/js'))
-	}
+	// get mode for webpack configuration file
+	const mode = process.env.NODE_ENV;
 
+	return gulp.src('src/js/site.js')
+		.pipe(webpackStream(config.WEBPACK_CONFIG(mode)))
+		.on('error', handleError)
+		.pipe(gulp.dest('dist/js'));
 });
 
-
-/*  
- * Javascript watch
+/**
+ * Image optimisation task
  */
-gulp.task('js-watch', ['js'], (done) => {
-	
-	browserSync.reload();
-	done();
-});
+gulp.task('images', () => gulp.src(STATIC_IMAGE_SRC)
+	.pipe(imagemin(config.IMAGMIN))
```

Comment:
Could the config here be inline to help with code readability – that'd go for all of them except where the config is used more than once.